codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/ShapeValueGenerator.java

```diff
@@ -0,0 +1,367 @@
+                        }
+
+                        memberShape = model.expectShape(member.getTarget());
+                        String memberName = symbolProvider.toMemberName(member);
+
+                        writer.write("$L: ", memberName);
+                        valueGen.writeShapeValueInline(writer, memberShape, valueNode);
+                        writer.write(",");
+                        break;
+
+                    case MAP:
+                        memberShape = model.expectShape(this.currentShape.asMapShape().get().getValue().getTarget());
+
+                        writer.write("$S: ", keyNode.getValue());
+                        valueGen.writeShapeValueInline(writer, memberShape, valueNode);
+                        writer.write(",");
+                        break;
+
+                    default:
+                        throw new CodegenException("unexpected shape type " + currentShape.getType());
+                }
+            });
+
+            return null;
+        }
+
+        @Override
+        public Void booleanNode(BooleanNode node) {
+            if (!currentShape.getType().equals(ShapeType.BOOLEAN)) {
+                throw new CodegenException("unexpected shape type " + currentShape + " for boolean value");
+            }
+
+            writer.writeInline("$L", node.getValue() ? "true" : "false");
+            return null;
+        }
+
+        @Override
+        public Void nullNode(NullNode node) {
+            if (currentShape.getType() == ShapeType.STRING && currentShape.hasTrait(EnumTrait.class)) {
+                writer.writeInline("\"\"");
+            } else {
+                writer.writeInline("nil");
+            }
+
+            return null;
+        }
+
+        @Override
+        public Void numberNode(NumberNode node) {
+            switch (currentShape.getType()) {
+                case BOOLEAN:
+                    writer.writeInline("$L", !node.getValue().equals(0) ? "true" : "false");
+                    break;
+
+                case BLOB:
+                    throw new CodegenException("unexpected number value targeting Blob shape " + currentShape);
+
+                case STRING:
+                    writer.writeInline("$S", node.getValue());
+                    break;
+
+                case TIMESTAMP:
+                    writer.addUseImports(GoDependency.SMITHY_TIME);
+                    writer.writeInline("smithytime.ParseEpochSeconds($L)", node.getValue());
+                    break;
+
+                case BYTE:
+                case SHORT:
+                case INTEGER:
+                case LONG:
+                case FLOAT:
+                case DOUBLE:
+                    writer.writeInline("$L", node.getValue());
+                    break;
+
+                case BIG_INTEGER:
+                    writer.addUseImports(GoDependency.BIG);
+                    writer.writeInline("func() *big.Int {\n"
```

Comment:
Looks like the logic here and for `BIG_DECIMAL` is duplicated with the `stringNode` method, consider abstracting.